apps/test/unit/code-studio/components/progress/StatusProgressDotTest.js

```diff
@@ -0,0 +1,217 @@
+import React from 'react';
+import { assert } from 'chai';
+import { shallow } from 'enzyme';
+import sinon from 'sinon';
+
+import { StatusProgressDot } from
+  '@cdo/apps/code-studio/components/progress/StatusProgressDot';
+import * as stageLockRedux from '@cdo/apps/code-studio/stageLockRedux';
+import { LevelStatus } from '@cdo/apps/code-studio/activityUtils';
+import { SignInState } from '@cdo/apps/code-studio/progressRedux';
+
+const ViewType = stageLockRedux.ViewType;
+
+function statusFromWrapper(wrapper) {
+  // StatusProgressDot returns a (connected) ProgressDot. This represents the
+  // prop passed to ProgressDot
+  return wrapper.node.props.status;
+}
+
+describe('StatusProgressDot', () => {
+  const fullyLockedStageId = 123;
+  const partiallyLockedStageId = 321;
+
+  before(() => {
+    sinon.stub(stageLockRedux, 'fullyLockedStageMapping', () => ({
+      [fullyLockedStageId]: true,
+      [partiallyLockedStageId]: false
+    }));
+  });
+  after(() => {
+    stageLockRedux.fullyLockedStageMapping.restore();
+  });
+
+  describe('handling lockable assessments', () => {
+    it('overrides status to be locked when viewing fully locked section as student', () => {
+      const wrapper = shallow(
+        <StatusProgressDot
+          level={{
+            icon: null,
+            ids: [5275],
+            kind: 'assessment',
+            next: [2, 1],
+            position: 1,
+            previous: false,
+            status: LevelStatus.perfect,
+            title: 1,
+            uid: '5275_0',
+            url: '/test-url'
+          }}
+          courseOverviewPage={true}
+          stageId={fullyLockedStageId}
+          viewAs={ViewType.Student}
+          showProgress={true}
+          saveAnswersBeforeNavigation={false}
+        />
```

Comment:
I might factor out generation of the common props being used across all of these tests into a single place so that it is easier to see what is changing from one test to the next.